rasa/core/tracker_store.py

```diff
@@ -558,11 +596,46 @@ def retrieve(self, sender_id: Text) -> Optional[DialogueStateTracker]:
             return
 
         events = self._events_from_serialized_tracker(stored)
-        if not self.load_events_from_previous_conversation_sessions:
+        if not fetch_events_from_all_sessions:
             events = self._events_since_last_session_start(events)
 
+        return events
+
+    def retrieve(self, sender_id: Text) -> Optional[DialogueStateTracker]:
+        """Retrieves tracker for the latest conversation session.
+
+        Args:
+            sender_id: the message owner ID
+
+        Returns:
+            `DialogueStateTracker` containing the latest conversation session.
+        """
+        # TODO: Remove this in Rasa Open Source 3.0 along with the
+        # deprecation warning in the constructor
+        if self.retrieve_events_from_previous_conversation_sessions:
+            return self.retrieve_full_tracker(sender_id)
+
+        events = self._retrieve(sender_id, fetch_events_from_all_sessions=False)
+
         return DialogueStateTracker.from_dict(sender_id, events, self.domain.slots)
 
+    def retrieve_full_tracker(
+        self, conversation_id: Text
+    ) -> Optional[DialogueStateTracker]:
+        """Retrieves tracker containing all conversation sessions.
```

Comment:
Optional: Imo it's not required to re-specify docstrings for methods of the abstract base class which are overriden.